Add newer sota-client with rvi and genivi-swm

alexhumphreys opened the pull request:
[GDP-183] https://at.projects.genivi.org/~/GDP-183

To allow genivi SOTA to support GDP, the following changes were made:

- Add rvi.bbappend to start RVI using systemd and configure it to use sota.genivi.org backend
- Add rvi-sota-client_git.bbappend for to configure it communicate with RVI and genivi-swm
- Add genivi-swm recipe, manages package updates, updates to ECUs etc. It is required by rvi-sota-client when using RVI. 
- Add python-storm recipe, a dependency of genivi-swm
- Downgrade python-pygobject 3.18.2 -> 2.28.3 as import gobject is failing with 3.18.2 
from poky/meta layer

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+LIC_FILES_CHKSUM = \
+"file://${S}/LICENSE;md5=8d2127b230519ce4ad2c16070bdeb7b3"
+
+SRCREV = "${AUTOREV}"
```
I don't like to use AUTOREV. Please use specific commit hash value

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+
+SRCREV = "${AUTOREV}"
+SRC_URI = "\
+  git://github.com/advancedtelematic/genivi_swm;branch=gdp-build;protocol=https \
```
Use 4 spaces

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+  "
+FILESEXTRAPATHS_prepend := "${THISDIR}:"
+
+PR="1"
```
PR = "r0" or not required.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+#inherit externalsrc
+#EXTERNALSRC = "${THISDIR}/../../../../../../../work/genivi_swm"
+
+FILESEXTRAPATHS_prepend := "${THISDIR}:"
```
Not required. please add your service files under "genivi-swm/genivi-swm or genivi-swm/files" directory.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+SYSTEMD_SERVICE_${PN} = " package_manager.service partition_manager.service module_loader_ecu1.service software_loading_manager.service lifecycle_manager.service "
+
+#inherit externalsrc
+#EXTERNALSRC = "${THISDIR}/../../../../../../../work/genivi_swm"
```
Please remove two lines.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+  file://module_loader_ecu1.service \
+  file://org.genivi.SoftwareLoadingManager.conf \
+  "
+FILESEXTRAPATHS_prepend := "${THISDIR}:"
```
Please remove this line and you need to move your service and conf files under "genivi-swm/genivi-swm or genivi-swm/files" directory.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+SECTION = "base"
+LICENSE = "Proprietary"
+LIC_FILES_CHKSUM = \
+"file://${S}/LICENSE;md5=8d2127b230519ce4ad2c16070bdeb7b3"
```
Use one line

jeremiah replied:
What is the "Proprietary" license here? Do we have permission to include this in an open source project?

alexhumphreys replied:
My bad, the project is MPL-2.0.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+  /etc/dbus-1/system.d/org.genivi.SoftwareLoadingManager.conf \
+  "
+
+DEPENDS = " systemd"
```
DEPENDS = "systemd"

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+  install -d ${D}${exec_prefix}/lib/genivi-swm/
+
+  # TODO: Probably want to just copy in the needed files
+  cp -r ${S}/* ${D}${exec_prefix}/lib/genivi-swm/
```
${exec_prefix}/lib/genivi-swm -> ${libdir}/${PN}

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,67 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}:"
+
+do_install () {
+  install -d ${D}${exec_prefix}/lib/genivi-swm/
```
${exec_prefix}/lib/genivi-swm -> ${libdir}/${PN}

chbae commented on meta-genivi-dev/recipes-sota/python-pygobject/python-pygobject_2.28.3.bb:
```diff
@@ -0,0 +1,55 @@
+# Copied from http://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-devtools/python?h=jethro d462015
```
Why do you downgrade this component?

alexhumphreys replied:
Honestly, I hadn't spotted the poky/meta layer in there, I had only seen the meta-* layers in the root directory. However, I tried removing this python-pygobject_2.28.3.bb and using the python-pygobject_3.18.2 from poky, and from the python prompt I was unable to run `import gobject`.

```
root@qemux86-64:~# python
Python 2.7.11 (default, Jan 27 2017, 13:16:50) 
[GCC 5.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import gobject
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named gobject
```

So I guess something else has changed with the upgrade?

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+
```
meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend  -> meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bbappend

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+
+PR="r5"
```
PR="r5" -> EXTENDPRAUTO = ".1" or please remove it.

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+
+PR="r5"
+FILESEXTRAPATHS_prepend := "${THISDIR}:"
```
FILESEXTRAPATHS_prepend := "${THISDIR}:" -> FILESEXTRAPATHS_prepend := "${THISDIR}:files" and move your toml and service files to "/files" directory

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+  file://org.genivi.SotaClient.conf \
+	"
+
+do_install_append () {
```
please use 4 spaces

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+SRC_URI += "\
+	file://sota.toml \
+	file://sota_client.service \
+  file://org.genivi.SotaClient.conf \
```
please use 4 spaces

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+
+do_install_append () {
+  cd "${B}"
+  ls ..
```
please don't use cd and ls command here

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+}
+
+FILES_${PN} += " \
+  /etc/sota.toml \
```
etc -> "${sysconfdir}

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client-append/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,23 @@
+
+FILES_${PN} += " \
+  /etc/sota.toml \
+  /etc/dbus-1/system.d/org.genivi.SotaClient.conf \
```
etc -> "${sysconfdir}

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,34 @@
+PR="r6"
+FILESEXTRAPATHS_prepend := "${THISDIR}:"
```
FILESEXTRAPATHS_prepend := "${THISDIR}:" -> FILESEXTRAPATHS_prepend := "${THISDIR}:files" and move your toml and service files to "/files" directory

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,34 @@
+PR="r6"
```
PR="r6" -> EXTENDPRAUTO = ".1" or please remove it.

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,34 @@
+
+do_install_append () {
+  cd "${B}"
+  ls ..
```
please don't use cd and ls command in here

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,34 @@
+}
+
+FILES_${PN} += " \
+  /usr/bin/run_rvi.sh \
```
is it really required? may be not.

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,34 @@
+INITSCRIPT_NAME = ""
+INITSCRIPT_PARAMS = ""
+
+SYSTEMD_AUTO_ENABLE ??= "enable"
```
why do you use ??(default value) ?

chbae commented on meta-genivi-dev/recipes-sota/squashfs-tools/squashfs-tools_git.bb:
```diff
@@ -0,0 +1,38 @@
+# Note, we can probably remove the lzma option as it has be replaced with xz,
```
squashfs-tools_git is already existed in poky. Why do you add in here?

chbae commented:
@alexhumphreys 
Thanks for your contribution but you need to change a lot.

zeenix reviewed (changes requested):
While @chbae & @jeremiah already did a good job of reviewing the actual changes, I would like to remind you of requirement 4 of Patch Process: https://at.projects.genivi.org/wiki/display/PROJ/How+to+contribute+to+GENIVI

Good work on providing lots of details in commit log though.

While for small fixes etc, I wouldn't ask for this but in this case, it'd be nice to have a Jira ticket attached to this. If one doesn't exist, please create one and ensure it's referred in the commit log.

alexhumphreys commented:
@chbae thanks for the feedback, I don't use bitbake too often so I'm pretty shaky. I'm just testing a build that addresses most of these then I'll update the PR. 
@zeenix I'm not sure if there's a Jira ticket, but I'll create one.

alexhumphreys commented:
@zeenix ah, turns out there is a ticket https://at.projects.genivi.org/jira/browse/GDP-183, I'll add this to the commit log.

zeenix commented:
Actually being a perfectionist, I'd like some improvements to commit log details as well :)

1. Would be nice to get some details on the benefits from this change.

2. We prefer smaller commits and this patch is updating/adding a lot. Could it not be divided into smaller more patches (inter-dependency between patches is fine, as long as it's linear and no commit breaks build on it's own).

> Added for gdp challenge work:

Not sure what is meant by "gdp challenge work" here, so would be good to be more specific here. etc. Also "GDP" instead of "gdp".

>  - genivi-swm

Some description of genivi-swm would be nice and why we are adding it.

>  - rvi (appends for rvi recipe)

It's not obvious what that means. It should be possible to decipher the commit log message without actually looking at the changes.

> Copied from projects ats doesn't maintain:

"ats" -> "ATS".

zeenix commented:
@alexhumphreys But please don't get me wrong, your contribution is more than welcome! :) I have two general questions as well:

1. Looking at your changes, I understand that this is not the new C++ one but is that not expected to be available soon?

2. I forgot. :)

Commit:
rvi: Start RVI using systemd

The systemd service file configures RVI to use the backend located at
sota.genivi.org.

Signed-off-by: Alex Humphreys <alex@advancedtelematic.com>

Commit:
rvi-sota-client: configure to use RVI

rvi-sota-client supports two methods of communication. The first is
http, the second is RVI which requires the genivi-swm. This change
configures rvi-sota-client to use the latter method, rather than
previously where there was no default configuration.

Signed-off-by: Alex Humphreys <alex@advancedtelematic.com>

alexhumphreys commented:
@chbae I've addressed most of the things you mentioned, if there's any more just let me know.

@zeenix no worries, a decent git log is a worthy goal, plus rewriting history is fun :) 

There's now multiple commits. Each has an explanation of what's going on in the commit message, and each builds, though I didn't try running each. The message at the top of this for the PR has been reworded to add the Jira ticket and to explain what's going on a bit more. Should the ticket also be in each commit message, or is it ok if it's just in the merge commit?

Let me know if they need any more changes and I'll be happy to oblige.

alexhumphreys commented:
@zeenix oh, and yes. This is not the new c++ client. That should be arriving soon, but I'm not on that project so I'm not sure what the timeline is. This work was part of the first milestone of the SOTA GENIVI Challenge work. Think a working end to end was needed integration, that way the new c++ client (which is part of the second milestone maybe?) could be just dropped in and it would be clear if it was working.

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,22 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+SRC_URI += "\
```
please use 4 spaces

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,22 @@
+
+do_install_append () {
+    install -d "${D}${systemd_unitdir}/system"
+    install -m 0644 "../rvi.service" "${D}${systemd_unitdir}/system/"
```
../rvi.service -> ${WORKDIR}/rvi.service

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,22 @@
+do_install_append () {
+    install -d "${D}${systemd_unitdir}/system"
+    install -m 0644 "../rvi.service" "${D}${systemd_unitdir}/system/"
+    install -m 0644 "../device_id" "${D}${sysconfdir}/opt/rvi/device_id"
```
../device_id -> ${WORKDIR}/device_id

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+
+FILES_${PN} = " \
+    /usr/lib/genivi-swm/* \
+    /lib/systemd/system/package_manager.service \
```
/lib/systemd/system -> ${systemd_system_unitdir}

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+S = "${WORKDIR}/git"
+
+FILES_${PN} = " \
+    /usr/lib/genivi-swm/* \
```
/usr/lib/genivi-swm -> ${libdir}/${PN}

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+    /lib/systemd/system/partition_manager.service \
+    /lib/systemd/system/module_loader_ecu1.service \
+    /lib/systemd/system/software_loading_manager.service \
+    /lib/systemd/system/lifecycle_manager.service \
```
/lib/systemd/system -> ${systemd_system_unitdir}

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+DEPENDS = "systemd"
+
+RDEPENDS_${PN} += " \
+    python-subprocess \
```
RDEPENDS_${PN} += -> RDEPENDS_${PN} = "

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+    "
+
+inherit systemd
+SYSTEMD_SERVICE_${PN} = " package_manager.service partition_manager.service module_loader_ecu1.service software_loading_manager.service lifecycle_manager.service "
```
please remove empty space before and after ""

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+    cp -r ${S}/* ${D}${libdir}/${PN}
+
+    install -d ${D}${systemd_unitdir}/system
+    install -c "../package_manager.service" ${D}${systemd_unitdir}/system
```
../package_manager.service ${D}${systemd_unitdir}/system  -> ${WORKDIR}/package_manager.service ${D}${systemd_system_unitdir}

And please change below lines

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+LICENSE = "LGPL-2.1"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=e91355d8a6f8bd8f7c699d62863c7303"
+
+SRC_URI = "https://launchpad.net/storm/trunk/0.20/+download/storm-0.20.tar.gz \
```
please use one line.

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+
+SRC_URI = "https://launchpad.net/storm/trunk/0.20/+download/storm-0.20.tar.gz \
+  "
+SRC_URI[md5sum] = "8628503141f0f06c0749d607ac09b9c7"
```
Why don't you add SRC_URI[sha256sum] as well?

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+
+inherit setuptools
+
+DEPENDS = "python "
```
This components has build time dependency? Not run time dependency?
If you need this, please use DEPENDS = "python"

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+inherit setuptools
+
+DEPENDS = "python "
+RDEPENDS_${PN} += " python-setuptools \
```
RDEPENDS_${PN} = "python....

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+
+DEPENDS = "python "
+RDEPENDS_${PN} += " python-setuptools \
+                    python-pkgutil \
```
Use 4 spaces

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+do_compile_prepend() {
+    BUILD_SYS=${BUILD_SYS} HOST_SYS=${HOST_SYS} \
+    ${STAGING_BINDIR_NATIVE}/python-native/python setup.py install || \
+    true
```
please integrate to line 22

chbae commented on meta-genivi-dev/recipes-sota/python-storm/python-storm_0.20.bb:
```diff
@@ -0,0 +1,37 @@
+export STAGING_LIBDIR
+
+FILES_${PN} = " \
+  /usr/lib/python2.7/site-packages \
```
use 4 spaces.
/usr/lib -> ${libdir}
python2.7 -> ${PYTHON_DIR} right? please check this again.

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,19 @@
+
+do_install_append () {
+    install -d "${D}${sysconfdir}"
+    install -c "../sota.toml" "${D}${sysconfdir}"
```
../sota.toml -> ${WORKDIR}/sota.toml

please change below lines as well.

chbae commented on meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bbappend:
```diff
@@ -0,0 +1,19 @@
+    "
+
+do_install_append () {
+    install -d "${D}${sysconfdir}"
```
install -d "${D}${sysconfdir}" -> install -d ${D}${sysconfdir}

zeenix commented:
> There's now multiple commits. Each has an explanation of what's going on in the commit message, and each builds, though I didn't try running each.

Awesome. The important thing is to ensure that no commit depends on changes that come after it.

> The message at the top of this for the PR has been reworded to add the Jira ticket and to explain what's going on a bit more. Should the ticket also be in each commit message, or is it ok if it's just in the merge commit?

Great! Yes, please and also please add 'Signed-off-by: ' to each commit.

Thanks for your understand and patience but I'll have to test it a bit more. :)

zeenix reviewed (changes requested):
The description part of commit logs is supposed to be normal English and hence follows the usual English grammar as much as possible/makes sense:

* '.' after sentences please.
* Empty line between paragraphs.

Although in this particular case, I don't think "Add rvi.bbappend" is the most relevant bit but "Start RVI using systemd" as former is implementation detail. So I'd suggest  "Start RVI using systemd" as the commit summary/shortlog.

zeenix reviewed (changes requested):
* Same comment about '.' at end of sentences and empty lines between paragraphs. Oh and please don't break to newline for starting a new sentence but to only ensure that no line goes beyond 74 characters.

* We have the convention of adding 'prefix: ' to commit shortlog to indicate which module is being changed/added.

* A tip for ensuring shortlog remains short (50 chars): make use of "&". :) I think this is mentioned in the guidelines I pointed to as well: https://wiki.gnome.org/Git/CommitMessages

* Talking of that guideline, there is a "Checklist" in there. If you go through that checklist for each of your commits, it'd save us both a lot of time. :) Just a suggestion.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+    file://org.genivi.SoftwareLoadingManager.conf \
+    "
+
+PR="r0"
```
PR = "r0"

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,22 @@
+    "
+
+do_install_append () {
+    install -d "${D}${systemd_unitdir}/system"
```
${systemd_unitdir}/system -> ${systemd_system_unitdir}

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,22 @@
+
+do_install_append () {
+    install -d "${D}${systemd_unitdir}/system"
+    install -m 0644 "${WORKDIR}/rvi.service" "${D}${systemd_unitdir}/system/"
```
${systemd_unitdir}/system -> ${systemd_system_unitdir}

chbae commented on meta-genivi-dev/recipes-sota/rvi/rvi_git.bbappend:
```diff
@@ -0,0 +1,22 @@
+INITSCRIPT_NAME = ""
+INITSCRIPT_PARAMS = ""
+
+pkg_postinst_${PN} () {
```
why this function is required?

alexhumphreys replied:
The rvi package generates a random uuid after it is installed. This function overwrites that so instead nothing happens. This way you can add your device id to `recipes-sota/rvi/files/device_id` and it won't get blitzed.

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+    # TODO: Probably want to just copy in the needed files
+    cp -r ${S}/* ${D}${libdir}/${PN}
+
+    install -d ${D}${systemd_unitdir}/system
```
${systemd_unitdir}/system -> ${systemd_system_unitdir}

chbae commented on meta-genivi-dev/recipes-sota/genivi-swm/genivi-swm_git.bb:
```diff
@@ -0,0 +1,60 @@
+    install -c "${WORKDIR}/partition_manager.service" ${D}${systemd_unitdir}/system
+    install -c "${WORKDIR}/module_loader_ecu1.service" ${D}${systemd_unitdir}/system
+    install -c "${WORKDIR}/software_loading_manager.service" ${D}${systemd_unitdir}/system
+    install -c "${WORKDIR}/lifecycle_manager.service" ${D}${systemd_unitdir}/system
```
${systemd_unitdir}/system -> ${systemd_system_unitdir}

chbae commented on meta-genivi-dev/recipes-sota/python-pygobject/python-pygobject_2.28.3.bb:
```diff
@@ -0,0 +1,55 @@
+LIC_FILES_CHKSUM = "file://COPYING;md5=a916467b91076e631dd8edb7424769c7"
+DEPENDS = "python python-pygobject-native libffi glib-2.0"
+DEPENDS_class-native = "python-native libffi-native glib-2.0-native"
+RDEPENDS_class-native = ""
```
I thought that this is not required right?

alexhumphreys replied:
@chbae Not sure if you saw my earlier comment, but I tried removing this python-pygobject_2.28.3.bb recipe, and so it would use the upgraded one from poky/meta. However, the genivi-swm was unable to import it. When I ran a python prompt and tried `import gobject` I got the same error. Has something else changed with this package between versions? Do you know if I need to configure something differently to use it?

chbae reviewed (changes requested):
@alexhumphreys 
Good work. just my opinion and I'll test on rpi3 tomorrow.
Could you let me know how to test it?

1. Why don't you integrate "recipes-sota: Add genivi-swm and dependencies" and "	recipes-sota: add swm to packagegroup-gdp-rvi" to one commit "Add genivi-swm and dependencies". Not required recipes-sota prefix.

2. typo in commit message of "recipes-sota: add swm to packagegroup-gdp-rvi"
perferm -> perform.

alexhumphreys commented:
@chbae I've updated the commits to match your points 1 and 2.

For testing, you can build an image and run it and check if the relevant processes are running. you can check that with:

```
systemctl status rvi
systemctl status software_loading_manager
systemctl status sota_client
```

To test installing a package, well, that's a bit more involved. You'll need to
- log in at https://sota.genivi.org/, with some GENIVI credentials
- create a package, using the package format from: https://github.com/GENIVI/genivi_swm
- create a vehicle
- copy the uuid of the vehicle from the url bar
- add that uuid to `recipes-sota/rvi/files/device_id` and `recipes-sota/rvi-sota-client/files/sota.toml`
- build and start the image
- go back to https://sota.genivi.org/#/packages, and in the packages list click Create Campaign
- create a campaign
- check your device to see if your package was installed

We're currently writing up proper documentation for how all of this will work. I'll add the link to the documentation when it's available.

chbae commented:
build is ok for rpi3 but I didnt run.
Looks fine to me.

zeenix reviewed (commented):
Thanks so much for reworking your patches. We are almost there. Some comments still:

* Aren't these two changes? "Also" in commit log typically indicates that and each logical change in a separate patch.

* Would be really good if log also says what is the client being ported *from* and not just *to*. E.g "..communicate over D-Bus with genivi-swm rather than.."

zeenix reviewed (changes requested):
Similar comment here: Dependencies could be added in a separate patch so it should be. Smaller the patches, the easier it is to manage them later (e.g reverting).

zeenix reviewed (changes requested):
Similar comment here: Change of sota.genivi.org is an independent change.

alexhumphreys commented:
@zeenix let's take this one commit at a time:

- rvi-sota-client: configure to use RVI
I don't believe that configuring rvi-sota-client to use RVI and dbus is two changes. rvi-sota-client supports 2 methods of communication, the first is http, the second is RVI. When using RVI, genivi-swm is required. The way rvi-sota-client communicates with genivi-swm is via dbus. So it doesn't make sense to split this into two changes. I rewrote the commit message to reflect this.

- Add genivi-swm and dependencies
You're right that this could be split into two commits. And it was split into two commits, until your fellow contributor @chbae suggested in a previous comment that they should be merged. I'm new to this project so I'm not sure which one of you has priority here, but perhaps you can come to an agreement with each other before I change this again.

- rvi: Start RVI using systemd 
I disagree with the statement "Change of sota.genivi.org is an independent change". There wasn't a systemd service file before, now there is systemd service file, and one of the lines of that service file is an environment variable that points to sota.genivi.org. I rewrote the commit message to reflect this.

codders commented:
@zeenix At this point it seems that we're spending an unreasonable amount of resources trying to get something merged. Can you please either accept the pull request, or make an alternative pull request that meets your requirement? Or should we continue to devote development time to this?

jeremiah commented:
I don't think its unreasonable in and of itself, GENIVI asks for smaller patches and pull requests as that is a best practice (not least for the reasons Zeeshan states.) Also, spending the time to do this now means that we don't have to spend time to de-couple the commits should we need to revert something. 

Of course, if there are deadlines involved, perhaps we opt for moving quickly.

zeenix commented:
@codders I understand you have deadlines and want to move forward quickly with least amount of time spent. I'd suggest you should see this more of learning opportunity since all well-established and good quality open source projects out there will ask you to split your patches into logical chunks. I can certainly help with perfecting the commit log messages.

zeenix commented:
@alexhumphreys Firstly, sorry for the late reply:

> @zeenix let's take this one commit at a time:

Sure.

> rvi-sota-client: configure to use RVI
> I don't believe that configuring rvi-sota-client to use RVI and dbus is two changes. rvi-sota-client supports 2 methods of communication, the first is http, the second is RVI. When using RVI, genivi-swm is required.

Ah ok, in that cast you just need to word the commit message details to reflect that, so it doesn't sound like two different changes.

> Add genivi-swm and dependencies
> You're right that this could be split into two commits. And it was split into two commits, until your fellow contributor @chbae suggested in a previous comment that they should be merged. I'm new to this project so I'm not sure which one of you has priority here, but perhaps you can come to an agreement with each other before I change this again.

Ouch, so sorry about that. When I became the maintainer, I introduced new policies and I guess this is one that I didn't advertise well enough. I'll talk to @chbae and ensure he understands the new policy. Please do split the patch, as you originally did. 

> rvi: Start RVI using systemd
> I disagree with the statement "Change of sota.genivi.org is an independent change". There wasn't a systemd service file before, now there is systemd service file, and one of the lines of that service file is an environment variable that points to sota.genivi.org. I rewrote the commit message to reflect this.

Cool, thanks.

zeenix commented:
@codders Besides, based on my reply to @alexhumphreys, we're almost there. :)

Commit:
recipes-sota: add genivi-swm dependencies

Add the dependencies required for genivi-swm. It requires python-storm
as a database orm. Along with that it requires a previous version of
python-pygobject, version 2.28.3.

Signed-off-by: Alex Humphreys <alex@advancedtelematic.com>

Commit:
recipes-sota: add genivi-swm recipe

Adds the recipe for genivi-swm. It is installed as part of the
packagegroup-gdp-rvi.

Signed-off-by: Alex Humphreys <alex@advancedtelematic.com>

alexhumphreys commented:
@zeenix I have split the dependencies out into their own commit. I have updated the commit messages for each commit. Take a look and see if they are correct, if you want any changes to the commit messages please let me know. Ideally in very specific terms so that I don't have to change them again.

zeenix reviewed (commented):
I actually meant, each dependency as separate commit but I can split for you since that should be easy.

zeenix commented:
@alexhumphreys I split the patch for you and fixed a few small things too. Please do look through and compare with your tree and commit messages for future reference. :) Seems there is no way of telling github I've pushed the changes. :(
